src/apps/eos/actions/layout.py

```diff
@@ -0,0 +1,404 @@
+    fields.append("Code:")
+    fields.append(helpers.eos_name_to_string(msg.code))
+    fields.append("Type:")
+    fields.append(helpers.eos_name_to_string(msg.type))
+    fields.append("Requirement:")
+    fields.append(helpers.eos_name_to_string(msg.requirement))
+
+    pages = list(chunks(fields, FOUR_FIELDS_PER_PAGE))
+    paginator = paginate(show_lines_page, len(pages), FIRST_PAGE, pages, text)
+
+    await ctx.wait(paginator)
+
+
+async def confirm_action_unlinkauth(ctx, msg: EosActionUnlinkAuth):
+    await ctx.call(
+        ButtonRequest(code=ButtonRequestType.ConfirmOutput), MessageType.ButtonAck
+    )
+
+    text = "Unlink Auth"
+    fields = []
+    fields.append("Account:")
+    fields.append(helpers.eos_name_to_string(msg.account))
+    fields.append("Code:")
+    fields.append(helpers.eos_name_to_string(msg.code))
+    fields.append("Type:")
+    fields.append(helpers.eos_name_to_string(msg.type))
+
+    pages = list(chunks(fields, FOUR_FIELDS_PER_PAGE))
+    paginator = paginate(show_lines_page, len(pages), FIRST_PAGE, pages, text)
+
+    await ctx.wait(paginator)
+
+
+async def confirm_action_newaccount(ctx, msg: EosActionNewAccount):
+    await ctx.call(
+        ButtonRequest(code=ButtonRequestType.ConfirmOutput), MessageType.ButtonAck
+    )
+
+    text = "New Account"
+    fields = []
+    fields.append("Creator:")
+    fields.append(helpers.eos_name_to_string(msg.creator))
+    fields.append("Name:")
+    fields.append(helpers.eos_name_to_string(msg.name))
+    fields += authorization_fields(msg.owner)
+    fields += authorization_fields(msg.active)
+
+    pages = list(chunks(fields, FOUR_FIELDS_PER_PAGE))
+    paginator = paginate(show_lines_page, len(pages), FIRST_PAGE, pages, text)
+
+    await ctx.wait(paginator)
+
+
+async def confirm_action_unknown(ctx, action, checksum):
+    await ctx.call(
+        ButtonRequest(code=ButtonRequestType.ConfirmOutput), MessageType.ButtonAck
+    )
+    text = "Arbitrary data"
+    fields = []
+    fields.append("Contract:")
+    fields.append(helpers.eos_name_to_string(action.account))
+    fields.append("Action Name:")
+    fields.append(helpers.eos_name_to_string(action.name))
+
+    fields.append("Checksum: ")
+    fields += split_data(hexlify(checksum).decode("ascii"))
+
+    pages = list(chunks(fields, FIVE_FIELDS_PER_PAGE))
+    paginator = paginate(show_lines_page, len(pages), FIRST_PAGE, pages, text)
+
+    await ctx.wait(paginator)
+
+
+@ui.layout
+async def show_lines_page(page: int, page_count: int, pages: list, header: str):
+    lines = ["%s" % field for field in pages[page]]
+    if header == "Arbitrary data":
+        text = Text("{} !!!".format(header), ui.ICON_WIPE, icon_color=ui.RED)
+    else:
+        text = Text("{}".format(header), ui.ICON_CONFIRM, icon_color=ui.GREEN)
+    text.mono(*lines)
+
+    content = Scrollpage(text, page, page_count)
+    if page + 1 == page_count:
+        await ConfirmDialog(content)
```

Comment:
You're ignoring the result here. See https://github.com/trezor/trezor-core/blob/master/src/apps/nem/mosaic/layout.py#L89

Reply:
Done.